src/worker/builder.rs

```diff
@@ -106,6 +107,30 @@ impl<E: 'static> WorkerBuilder<E> {
         self.register(kind, Closure(handler))
     }
 
+    /// Register a _blocking_ (synchronous) handler function for the given job type (`kind`).
+    ///
+    /// This is an analogue of [`register_fn`](WorkerBuilder::register_fn) for compute heavy tasks.
+    /// Internally, `tokio`'s `spawn_blocking` is used when a job arrives whose type matches `kind`,
+    /// and so the `handler` is executed in a dedicated pool for blocking operations. See `Tokio`'s
+    /// [docs](https://docs.rs/tokio/latest/tokio/index.html#cpu-bound-tasks-and-blocking-code) for
+    /// how to set the upper limit on the number of threads in the mentioned pool and other details.
+    ///
+    /// Note that it is not recommended to mix blocking and non-blocking tasks and so if many of your
+    /// handlers are blocking, you will want to launch a dedicated worker process (at least a separate
+    /// Tokio Runtime) where only blocking handlers will be used.
```

Comment:
I'm not sure I expect this to be necessary. As long as we're using `spawn_blocking`, it should be fine to have some async and some sync handlers.

Reply:
This is my reading of this [section](https://docs.rs/tokio/latest/tokio/index.html#cpu-bound-tasks-and-blocking-code) and namely this passage:
```
If your code is CPU-bound and you wish to limit the number of threads used to run it, you should use a separate thread pool dedicated to CPU bound tasks. For example, you could consider using the [rayon](https://docs.rs/rayon) library for CPU-bound tasks. It is also possible to create an extra Tokio runtime dedicated to CPU-bound tasks, but if you do this, you should be careful that the extra runtime runs only CPU-bound tasks, as IO-bound tasks on that runtime will behave poorly.
```
So "if many handlers are blocking" they _can_  employ one single runtime and mix blocking handlers with async ones, since we are using `spawn_blocking` (and so are protecting the cooperative scheduling for async tasks), but we are encouraging them to _consider_ launching a dedicated runtime for their blocking tasks. Or you think this note is redundant?

Reply:
I think giving a nudge in this direction is fine, although I think the nudge is currently too strong. I wonder if we just want to say something like:

> You can mix and match async and blocking handlers in a single `Worker`. However, note that there is no active management of the blocking tasks in tokio, and so if you end up with more CPU-intensive blocking handlers executing at the same time than you have cores, the asynchronous handler tasks (and indeed, all tasks) will suffer as a result. If you have a lot of blocking tasks, consider using the standard async job handler and add explicit code to manage the blocking tasks appropriately.

Reply:
updated